packages/next/next-server/server/next-server.ts

```diff
@@ -437,6 +442,9 @@ export default class Server {
     try {
       return await this.run(req, res, parsedUrl)
     } catch (err) {
+      if (this.onErrorMiddleware) {
+        this.onErrorMiddleware({ err, req, res })
+      }
```

Comment:
We should keep this logic in `logError` and pass it `req` / `res` instead.

Reply:
Fixed